### ISSUE TITLE
Feature: `headers.getFirst(header, returnValueIfNotExists = '')`

### ISSUE THREAD

**Issue by niftylettuce** [andris9/mailsplit#10]
Right now if you want to check if a header exists, a common approach is `const header = headers.getFirst('X-Some-Header');`, however if this header did not exist on the message, then it will still return an empty String of `''` - which may be misleading as the value should be null or undefined I think instead.  Perhaps we allow a second option of `returnValue`, which defaults to `''` for backwards compatibility, but if you pass it, if the header did not exist, it will return that value, e.g. `null` could be passed.

https://github.com/andris9/mailsplit/blob/master/lib/headers.js#L54-L56

**Comment by andris9** [comment]
I think that this method is good enough as it is. For header existence checking there should be some other method, eg `headers.exists(key)` or `header.has(key)` or something like that.

**Comment by niftylettuce** [comment]
That would be great actually.  By the way, I started sponsoring you.

**Comment by niftylettuce** [comment]
Should I submit a PR or did you want to add it?

Here's my current implementation:

```js
function hasHeader(headers, key) {
  if (!headers.parsed)
    headers._parseHeaders();
  key = headers._normalizeHeader(key);
  let header = headers.lines.find(line => line.key === key);
  return boolean(header);
}
```

**Comment by andris9** [comment]
> That would be great actually. By the way, I started sponsoring you.

Thats great, thank you!



> Should I submit a PR or did you want to add it?
> 
> Here's my current implementation:
> 
> ```js
> function hasHeader(headers, key) {
>   if (!headers.parsed)
>     headers._parseHeaders();
>   key = headers._normalizeHeader(key);
>   return typeof headers.lines.find(line => line.key === key) === 'string';
> }
> ```

Look good. Create a PR and I'll merge

**Comment by niftylettuce** [comment]
I'm going to submit a PR with this in just a moment

**Comment by niftylettuce** [comment]
Also note I had a typo in the above snippet, and have corrected it.  Instead of comparison `typeof x === 'string'` it should have been `typeof x === 'object'`, since a `.find` on the header lines will return an object `{ key: 'message-id', line: 'Message-ID: <abc@def>' }` match.

**Comment by niftylettuce** [comment]
@andris9 please accept PR https://github.com/andris9/mailsplit/pull/11 and then bump version and publish to npm.  you should not need to do a major nor minor bump, a patch should work OK since this is backwards compatible.  All tests passed, and I added new tests as well.

**Comment by andris9** [comment]
Even though there are no major changes I published it as v5.0.0 as I also upgraded some dependencies and these lately tend to have issues with older versions of Node. I don't even run tests on Node 6 anymore, so no idea if it still works or not.

**Comment by niftylettuce** [comment]
Thanks @andris9 as always, cheers
